frontend/src/pages/Chat/ChatRoom.tsx

```diff
@@ -0,0 +1,239 @@
+import { FormEvent, useContext, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { LogoutIcon, SendIcon } from '../../assets/icons/icons';
+import ChatList from '../../components/Chat/ChatList';
+import ChatMessage from '../../components/Chat/ChatMessage';
+import Loading from '../../components/commons/Loading';
+import Modal from '../../components/commons/Modal/Modal';
+import CustomInput from '../../components/CustomInput';
+import PageHeader from '../../components/PageHeader/PageHeader';
+import { UserInfoContext } from '../../context/UserInfoContext';
+import { useModal } from '../../hooks/useModal';
+import useSocket from '../../hooks/useSocket';
+import { remote } from '../../lib/api';
+import { useNavigate, usePathParams } from '../../lib/Router';
+import colors from '../../styles/colors';
+import { textMedium, textSmall } from '../../styles/fonts';
+import { ChatRoomInfo } from '../../types/chat';
+
+function Chat() {
+  const [message, setMessage] = useState('');
+  const { chatId } = usePathParams();
+  const [chatRoomInfo, setChatRoomInfo] = useState<ChatRoomInfo>(null!);
+  const userInfo = useContext(UserInfoContext);
+  const [partner, setPartner] = useState<string>('partner');
+  const { isModalOpen, openModal, closeModal } = useModal();
+  const navigate = useNavigate();
+
+  // const { data: chatRoomInfo } = useQuery<ChatRoomInfo>(
+  //   ['chat', chatId],
+  //   async () => {
+  //     const result = await remote(`/chat/${chatId}`);
+
+  //     console.log(result);
+  //     return result.data;
+  //   },
+  // );
+
+  useEffect(() => {
+    (async function () {
+      const result = await remote(`/chat/${chatId}`);
+
+      if (!result.data.roomInfo) {
+        navigate('/');
+      }
+
+      const { buyerId, sellerId } = result.data.roomInfo;
+      const partnerId = buyerId === userInfo.userId ? sellerId : buyerId;
+      const partner = await remote(`/auth/${partnerId}`);
+
+      setPartner(partner.data.nickname);
+      setChatRoomInfo(result.data);
+    })();
+  }, [chatId, userInfo]);
+
+  const { isLoading, sendMessage, receivedData } = useSocket(chatRoomInfo);
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    if (!message) return;
+    if (!chatRoomInfo?.roomInfo.buyerId) return;
+
+    sendMessage({
+      message,
+      senderId: userInfo.userId,
+    });
+
+    setMessage('');
+  };
+
+  const handleExitChatRoom = async () => {
+    await remote.delete(`/chat/${chatId}`);
+    navigate('/my');
+  };
+
```

Comment:
```suggestion
  useEffect(() => {
    if (chatListRef.current) {
      chatListRef.current.scrollTop = chatListRef.current.scrollHeight;
    }
  }, [receivedData]);
```
이런 코드 추가하면 채팅보내고, 처음 입장할 때 마지막 보낸 채팅이 보일 수 있을 것 같아용

Reply:
아 그러네요! 이 방법을 찾고 있었어요